contribs/documentation.py

```diff
@@ -84,25 +107,28 @@ class Converter:
 
     @classmethod
     def _scan_hint(cls, hint: type) -> dict[str, str]:
-        def recurse(type_: type, *metadata: Any) -> dict:
+        def recurse(
+            type_: type, *metadata: Any, **options: Unpack[AsyncAPIOptions]
+        ) -> dict:
             # Check if has a subtype i.e: type[subtype, ...]
             if origin_type := get_origin(type_):
                 subtype, *args = get_args(type_)
 
                 if origin_type is Annotated:
-                    return recurse(subtype, *args)
+                    return recurse(subtype, *args, **options)
 
                 elif origin_type is Literal:
                     return AsyncAPITypes.string(*args)
 
                 # i.e: Union[X, Y] or X | Y
                 elif origin_type in (UnionType, Union):
-                    return recurse(subtype)
+                    nullable = any([item is NoneType for item in args])
```

Comment:
Are you sure this works? nullable types use e.g. `str | None`. Wouldn't `None` be in `args`, instead of `NoneType`?